File: Core/Sources/HarmonicaCore/Entities/Playing/PlayingStyle.swift
public enum PlayingStyle: CaseIterable, Hashable, Sendable {
    case severalFingersSeveralNotes
    case severalFingersOneNote
    case oneFingerSeveralNotes

    public var coversTheContactWidth: Bool {
        self != .severalFingersOneNote
    }

    public var takesTheTopmostFingerOnly: Bool {
        self == .oneFingerSeveralNotes
    }
}

File: Web/src/domain/playing/playingStyle.ts
export type PlayingStyle =
    | 'severalFingersSeveralNotes'
    | 'severalFingersOneNote'
    | 'oneFingerSeveralNotes'

export const playingStyles: readonly PlayingStyle[] =
    ['severalFingersSeveralNotes', 'severalFingersOneNote', 'oneFingerSeveralNotes']

export function coversTheContactWidth(style: PlayingStyle): boolean {
    return style !== 'severalFingersOneNote'
}

export function takesTheTopmostFingerOnly(style: PlayingStyle): boolean {
    return style === 'oneFingerSeveralNotes'
}
